api/src/main/java/com/github/retrooper/packetevents/protocol/world/chunk/LightData.java

```diff
@@ -0,0 +1,162 @@
+package com.github.retrooper.packetevents.protocol.world.chunk;
+
+import com.github.retrooper.packetevents.manager.server.ServerVersion;
+import com.github.retrooper.packetevents.wrapper.PacketWrapper;
+
+import java.util.BitSet;
+
+public class LightData implements Cloneable {
+    private boolean trustEdges;
+    private BitSet blockLightMask;
+    private BitSet skyLightMask;
+    private BitSet emptyBlockLightMask;
+    private BitSet emptySkyLightMask;
+    private int skyLightCount;
+    private int blockLightCount;
+    private byte[][] skyLightArray;
+    private byte[][] blockLightArray;
+
+    public void read(PacketWrapper<?> packet) {
+        ServerVersion serverVersion = packet.getServerVersion();
+        if (serverVersion.isOlderThanOrEquals(ServerVersion.V_1_19_4)) {
+            trustEdges = packet.readBoolean();
+        }
+
+        skyLightMask = ChunkBitMask.readChunkMask(packet);
+        blockLightMask = ChunkBitMask.readChunkMask(packet);
+        emptySkyLightMask = ChunkBitMask.readChunkMask(packet);
+        emptyBlockLightMask = ChunkBitMask.readChunkMask(packet);
+
+        skyLightCount = serverVersion.isNewerThanOrEquals(ServerVersion.V_1_17) ? packet.readVarInt() : 18;
+        this.skyLightArray = new byte[skyLightCount][];
+        for (int x = 0; x < skyLightCount; x++) {
+            if (this.skyLightMask.get(x)) {
+                skyLightArray[x] = packet.readByteArray();
+            }
+        }
+
+        blockLightCount = serverVersion.isNewerThanOrEquals(ServerVersion.V_1_17) ? packet.readVarInt() : 18;
+        this.blockLightArray = new byte[blockLightCount][];
+        for (int x = 0; x < blockLightCount; x++) {
+            if (this.blockLightMask.get(x)) {
+                blockLightArray[x] = packet.readByteArray();
+            }
+        }
+    }
+
+    public void write(PacketWrapper<?> packet) {
```

Comment:
see above, for writes this would be `public static void write(PacketWrapper<?> packet, LightData data)` (in this case)